lib/oneview-sdk/image-streamer/resource/api300/artifact_bundle.rb

```diff
@@ -0,0 +1,175 @@
+        def extract(force = true)
+          ensure_uri
+          options = { 'Content-Type' => 'text/plain' }
+          response = @client.rest_put(@data['uri'] + "?extract=true&forceImport=#{force}", options)
+          @client.response_handler(response)
+          true
+        end
+
+        # Downloads the content of the selected artifact bundle to the admin's local drive.
+        # @param [String] local_drive_path Path to save file downloaded
+        # @return [Boolean] if file was downloaded
+        def download(local_drive_path)
+          raise IncompleteResource, "Missing required attribute 'downloadURI'" unless @data['downloadURI']
+          client.download_file(@data['downloadURI'], local_drive_path)
+        end
+
+        # Add a Build Plan to this ArtifactBundle
+        # @param [OneviewSDK::ImageStreamer::API300::BuildPlan] resource The BuildPlan resource with uri
+        # @param [TrueClass, FalseClass] read_only Indicates whether the BuildPlan will be readonly in artifact bundle package
+        # @raise [RuntimeError] if the BuildPlan uri is not set or it is not valid
+        def add_build_plan(resource, read_only = true)
+          add_resource(resource, 'buildPlans', read_only)
+        end
+
+        # Add a Plan Script to this ArtifactBundle
+        # @param [OneviewSDK::ImageStreamer::API300::PlanScript] resource The PlanScripts resource with uri
+        # @param [TrueClass, FalseClass] read_only Indicates whether the PlanScripts will be readonly in artifact bundle package
+        # @raise [RuntimeError] if the PlanScripts uri is not set or it is not valid
+        def add_plan_script(resource, read_only = true)
+          add_resource(resource, 'planScripts', read_only)
+        end
+
+        # Add a Deployment Plan to this ArtifactBundle
+        # @param [OneviewSDK::ImageStreamer::API300::DeploymentPlans] resource The DeploymentPlans resource with uri
+        # @param [TrueClass, FalseClass] read_only Indicates whether the DeploymentPlans will be readonly in artifact bundle package
+        # @raise [RuntimeError] if the DeploymentPlans uri is not set or it is not valid
+        def add_deployment_plan(resource, read_only = true)
+          add_resource(resource, 'deploymentPlans', read_only)
+        end
+
+        # Add a Golden Image to this ArtifactBundle
+        # @param [OneviewSDK::ImageStreamer::API300::GoldenImage] resource The GoldenImage resource with uri
+        # @param [TrueClass, FalseClass] read_only Indicates whether the GoldenImage will be readonly in artifact bundle package
+        # @raise [RuntimeError] if the GoldenImage uri is not set or it is not valid
+        def add_golden_image(resource, read_only = true)
+          add_resource(resource, 'goldenImages', read_only)
+        end
+
+        def self.ensure_resource!(resource)
+          raise "The resource #{resource.class} can not be retrieved. Ensure it have a valid URI." unless resource.retrieve!
```

Comment:
@aalexmonteiro Thinking better, incompleteResource makes more sense. What do yout think?
Jarred explained in other comment and convinced me.